Sync/Synchronizers/Bookmarks/ThreeWayTreeMerger.swift

```diff
@@ -607,6 +607,10 @@ class ThreeWayTreeMerger {
 
                 log.debug("Comparing local (\(local.localModified ??? "nil")) to remote (\(remote.serverModified)) clock for two-way value merge of \(guid).")
                 if local.localModified! > remote.serverModified {
+                    if let dateAddedRemote = remote.dateAdded,
```

Comment:
Add a comment:

```
// If we're taking the local record because it was modified, check to see
// whether the remote record has a creation date that we want to keep.
```

N.B., we also want to do exactly what we do on desktop: if there's no remote creation date, use the server modified time instead, if it's less than the local creation date. So I think this becomes:

```
let dateAddedRemote = remote.dateAdded ?? remote.serverModified
if (local.dateAdded ?? UInt64.max) > dateAddedRemote {
    return MergeState.new(…)
}
```